gluoncv/model_zoo/icnet.py

```diff
@@ -0,0 +1,381 @@
+    norm_kwargs : dict
+        Additional `norm_layer` arguments, for example `num_devices=4`
+        for :class:`mxnet.gluon.contrib.nn.SyncBatchNorm`.
+    pretrained_base : bool or str
+        Refers to if the FCN backbone or the encoder is pretrained or not. If `True`,
+        model weights of a model that was trained on ImageNet is loaded.
+
+
+    Reference:
+
+        Hengshuang Zhao, Xiaojuan Qi, Xiaoyong Shen, Jianping Shi and Jiaya Jia.
+        "ICNet for Real-Time Semantic Segmentation on High-Resolution Images." *ECCV*, 2018
+
+    Examples
+    --------
+    >>> model = ICNet(nclass=19, backbone='resnet50')
+    >>> print(model)
+    """
+    # pylint: disable=arguments-differ
+    def __init__(self, nclass, backbone='resnet50', aux=False, ctx=cpu(), pretrained_base=True,
+                 base_size=520, crop_size=480, **kwargs):
+        super(ICNet, self).__init__(nclass, aux=aux, backbone=backbone, ctx=ctx,
+                                    base_size=base_size, crop_size=crop_size,
+                                    pretrained_base=pretrained_base, **kwargs)
+
+        with self.name_scope():
+            self.maxpool = nn.MaxPool2D(pool_size=3, strides=2, padding=1, ceil_mode=True)
+
+            base_psp_head = _PSPHead(nclass,
+                                     feature_map_height=int(round(self._up_kwargs['height'] / 32)),
+                                     feature_map_width=int(round(self._up_kwargs['width'] / 32)),
+                                     **kwargs)
+            self.psp_head = nn.HybridSequential()
+            with self.psp_head.name_scope():
+                self.psp_head.add(
+                    base_psp_head.psp,
+                    base_psp_head.block[:-1]
+                )
+            self.psp_head.initialize(ctx=ctx)
+            self.psp_head.collect_params().setattr('lr_mult', 10)
+
+            self.head = _ICHead(nclass=nclass)
+            self.head.initialize(ctx=ctx)
+            self.head.collect_params().setattr('lr_mult', 10)
+
+            self.conv_sub1 = nn.HybridSequential()
+            with self.conv_sub1.name_scope():
+                self.conv_sub1.add(ConvBnRelu(3, 32, 3, 2, 1),
+                                   ConvBnRelu(32, 32, 3, 2, 1),
+                                   ConvBnRelu(32, 64, 3, 2, 1))
+            self.conv_sub1.initialize(ctx=ctx)
+            self.conv_sub1.collect_params().setattr('lr_mult', 10)
+
+            self.conv_sub4 = ConvBnRelu(512, 256, 1)
+            self.conv_sub4.initialize(ctx=ctx)
+            self.conv_sub4.collect_params().setattr('lr_mult', 10)
+
+            self.conv_sub2 = ConvBnRelu(512, 256, 1)
+            self.conv_sub2.initialize(ctx=ctx)
+            self.conv_sub2.collect_params().setattr('lr_mult', 10)
+
+    def hybrid_forward(self, F, x):
+        # sub 1
+        x_sub1_out = self.conv_sub1(x)
+
+        # sub_2
+        x_sub2 = F.contrib.BilinearResize2D(x, height=x.shape[2] // 2, width=x.shape[3] // 2)
```

Comment:
In order to use hybrid_forward, you cannot use `.shape` here. Please address it using something like `self._up_kwargs['height'] ` to pass in the shape.